dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/IndiciesInfo.java

```diff
@@ -0,0 +1,234 @@
+        }
+
+        public Builder setReindexWorking(String reindexWorking) {
+            this.reindexWorking = reindexWorking;
+            return this;
+        }
+
+        public Builder setSiteSearch(String siteSearch) {
+            this.siteSearch = siteSearch;
+            return this;
+        }
+
+        public IndiciesInfo build() {
+            return new IndiciesInfo(this);
+        }
+
+        public static Builder copy(final IndiciesInfo info) {
+            final IndiciesInfo.Builder builder = new IndiciesInfo.Builder();
+            builder.setWorking(info.getWorking());
+            builder.setLive(info.getLive());
+            builder.setReindexWorking(info.getReindexWorking());
+            builder.setSiteSearch(info.getSiteSearch());
+
+            return builder;
+        }
+    }
+
+    public static final SimpleDateFormat timestampFormatter = new SimpleDateFormat("yyyyMMddHHmmss");
+    public static final String CLUSTER_PREFIX = "cluster_";
+    private final static String INDEX_NAME_PATTERN = CLUSTER_PREFIX + "%s.%s_%s";
+
+    private final Map<IndexType, String> indiciesNames = new HashMap<>();
+
+    public IndiciesInfo(final Builder builder) {
+        this.indiciesNames.put(IndexType.LIVE, builder.live);
+        this.indiciesNames.put(IndexType.WORKING, builder.working);
+        this.indiciesNames.put(IndexType.REINDEX_LIVE, builder.reindexLive);
+        this.indiciesNames.put(IndexType.REINDEX_WORKING, builder.reindexWorking);
+        this.indiciesNames.put(IndexType.SITE_SEARCH, builder.siteSearch);
+    }
+
+    /**
+     * Return the current LIVE index name
+     * @return
+     */
+    public String getLive() {
+        return this.indiciesNames.get(IndexType.LIVE);
+    }
+
+    /**
+     * Return the current WORKING index name
+     * @return
+     */
+    public String getWorking() {
+        return this.indiciesNames.get(IndexType.WORKING);
+    }
+
+    /**
+     * Return the current REINDEX LIVE index name
+     * @return
+     */
+    public String getReindexLive() {
+        return this.indiciesNames.get(IndexType.REINDEX_LIVE);
+    }
+
+    /**
+     * Return the current REINDEX WORKING index name
+     * @return
+     */
+    public String getReindexWorking() {
+        return this.indiciesNames.get(IndexType.REINDEX_WORKING);
+    }
+
+    /**
+     * Return the current SITE SEARCH index name
+     * @return
+     */
+    public String getSiteSearch() {
+        return this.indiciesNames.get(IndexType.SITE_SEARCH);
+    }
+
+    /**
+     * Return the timestamp from a index name, a index name has the follow sintax:
+     *
+     * <b>cluster_<CLUSTER_ID>.<INDEX_NAME_PREFIX>_<TIME_STAMP></b>
+     * @return
+     */
+    public long getIndexTimeStamp(final IndexType indexType) {
+        Date startTime;
+        try {
+            final String indexName = indiciesNames.get(indexType);
+            final String indexTimestamp = indexName.substring(indexName.lastIndexOf("_") + 1);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)